app/controllers/support_interface/application_forms/immigration_status_controller.rb

```diff
@@ -0,0 +1,33 @@
+module SupportInterface
+  module ApplicationForms
+    class ImmigrationStatusController < SupportInterfaceController
+      before_action :set_application_form
+
+      def edit
+        @immigration_status_form = ImmigrationStatusForm.build_from_application(@application_form)
+      end
+
+      def update
+        @immigration_status_form = ImmigrationStatusForm.new(immigration_status_params)
+
+        if @immigration_status_form.save(@application_form)
+          redirect_to support_interface_application_form_path(@application_form)
+        else
+          redirect_to support_interface_application_form_edit_immigration_status_path(@application_form)
```

Comment:
Couldn't quite find a robust solution here. If I `render :edit` the view changes to the non eu version as `@immigration_status_form.eu_nationality?` must change to false for some reason.

Any ideas?

Reply:
I'd guess that on line 11 which does `ImmigrationStatusForm.new(immigration_status_params)` it also need to do `ImmigrationStatusForm.build_from_application(@application_form)` and then somehow merge in the params?

Reply:
Yep you do @stevehook. That was a temp fix until I figured it out. All fixed now.

Thanks both! ♻️